src/panels/config/lovelace/resources/dialog-lovelace-resource-detail.ts

```diff
@@ -0,0 +1,248 @@
+      this._url = "";
+      this._type = "module";
+    }
+    await this.updateComplete;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._params) {
+      return html``;
+    }
+    const urlInvalid = this._url.trim() === "";
+    // tslint:disable-next-line: prettier
+    const title = html`
+      ${this._params.resource
+        ? this._params.resource.url
+        : this.hass!.localize(
+            "ui.panel.config.lovelace.resources.detail.new_resource"
+          )}
+      <paper-icon-button
+        aria-label=${this.hass.localize(
+          "ui.panel.config.lovelace.resources.detail.dismiss"
+        )}
+        icon="hass:close"
+        dialogAction="close"
+        style="position: absolute; right: 16px; top: 12px;"
+      ></paper-icon-button>
+    `;
+    return html`
+      <ha-dialog
+        open
+        @closing="${this._close}"
+        scrimClickAction=""
+        escapeKeyAction=""
+        .heading=${title}
+      >
+        <div>
+          ${this._error
+            ? html`
+                <div class="error">${this._error}</div>
+              `
+            : ""}
+          <div class="form">
+            <h3 class="warning">
+              ${this.hass!.localize(
+                "ui.panel.config.lovelace.resources.detail.warning_header"
+              )}
+            </h3>
+            ${this.hass!.localize(
+              "ui.panel.config.lovelace.resources.detail.warning_text"
+            )}
+            <paper-input
+              .value=${this._url}
+              @value-changed=${this._urlChanged}
+              .label=${this.hass!.localize(
+                "ui.panel.config.lovelace.resources.detail.url"
+              )}
+              .errorMessage=${this.hass!.localize(
+                "ui.panel.config.lovelace.resources.detail.url_error_msg"
+              )}
+              .invalid=${urlInvalid}
+            ></paper-input>
+            <br />
+            <ha-paper-dropdown-menu
+              .label=${this.hass!.localize(
+                "ui.panel.config.lovelace.resources.detail.type"
+              )}
+            >
+              <paper-listbox
+                slot="dropdown-content"
+                .selected=${this._type}
+                @iron-select=${this._typeChanged}
+                attr-for-selected="type"
+              >
+                <paper-item type="module">
+                  ${this.hass!.localize(
+                    "ui.panel.config.lovelace.resources.types.module"
+                  )}
+                </paper-item>
+                <paper-item type="js">
+                  ${this.hass!.localize(
+                    "ui.panel.config.lovelace.resources.types.js"
+                  )}
+                </paper-item>
+                <paper-item type="css">
+                  ${this.hass!.localize(
+                    "ui.panel.config.lovelace.resources.types.css"
+                  )}
+                </paper-item>
+                <paper-item type="html">
```

Comment:
Let's drop this too. Only add it as a choice if the entry we're editing uses it. 

Maybe even better is to just deprecate it.

Reply:
#5014

Reply:
Hehe I already added the deprecated text :D